frontend/packages/ceph-storage-plugin/src/components/modals/clone-pvc-modal/clone-pvc-modal.tsx

```diff
@@ -0,0 +1,193 @@
+      metadata: {
+        name: clonePVCName,
+        namespace: _.get(resource, 'metadata.namespace'),
+      },
+      spec: {
+        storageClassName: _.get(resource, 'spec.storageClassName'),
+        dataSource: {
+          name: pvcName,
+          kind: 'PersistentVolumeClaim',
+          apiGroup: '',
+        },
+        accessModes: _.get(resource, 'spec.accessModes'),
+        resources: {
+          requests: {
+            storage: _.get(resource, 'spec.resources.requests.storage'),
+          },
+        },
+      },
+    };
+
+    handlePromise(k8sCreate(PersistentVolumeClaimModel, pvcCloneObj))
+      .then(() => {
+        setProgress(false);
+        close();
+      })
+      .catch((error) => {
+        setError(error);
+        setProgress(false);
+        setButton(true);
+        throw error;
+      });
+  };
+
+  const updateClonePVCName = (value: string) => {
+    setClonePVCName(value);
+  };
+
+  return (
+    <Form onSubmit={submit}>
+      <div className="modal-content modal-content--no-inner-scroll">
+        <ModalTitle>Clone</ModalTitle>
+        <ModalBody>
+          <FormGroup
+            label="Name"
+            isRequired
+            fieldId="clone-pvc-modal__name"
+            className="co-required"
+          >
+            <TextInput
+              isRequired
+              type="text"
+              id="clone-pvc-modal__name"
+              name="clone-pvc-modal__name"
+              value={clonePVCName}
+              onChange={updateClonePVCName}
+            />
+          </FormGroup>
+          <div className="clone-pvc-modal__details-section">
+            <p className="text-muted clone-pvc-modal__pvc-details">PVC Details</p>
+            <div className="clone-pvc-modal__details">
+              <div>
+                <div>
+                  <p className="clone-pvc-modal__pvc-details">Namespace</p>
+                  <p>
+                    <ResourceIcon kind="Namespace" />
+                    {resource.metadata.namespace}
+                  </p>
+                </div>
+                <div>
+                  <p className="clone-pvc-modal__pvc-details">Storage Class</p>
+                  <p>
+                    <ResourceIcon kind="StorageClass" />
+                    {resource.spec.storageClassName}
+                  </p>
+                </div>
+              </div>
+              <div>
+                <div>
+                  <p className="clone-pvc-modal__pvc-details">Requested Capacity</p>
+                  <p>
+                    {
+                      humanizeBinaryBytes(
+                        convertToBaseValue(resource.spec.resources.requests.storage),
+                      ).string
+                    }
+                  </p>
+                </div>
+                <div>
+                  <p className="clone-pvc-modal__pvc-details">Used Capacity</p>
+                  <p>{!queriesLoadError && humanizeBinaryBytes(pvcUsedCapacity).string}</p>
```

Comment:
What if there is an error ? I think you should handle that otherwise it will be blank.
Also, you might need to add loading state too.